Marshal.xcodeproj/project.pbxproj

```diff
@@ -427,7 +439,7 @@
 				PRODUCT_BUNDLE_IDENTIFIER = com.utahiosmac.Marshal;
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SKIP_INSTALL = YES;
-				SWIFT_VERSION = 3.0;
+				SWIFT_VERSION = 4.0;
```

Comment:
If we leave this at 3.0, but take the rest of the changes, Marshall will be compatible with both Xcode 8 and Xcode 9.

Reply:
That sounds perfect. @wgywgy would you make this change and then I'll get this merged in?

Reply:
I also get a warning with Swift4: `MarshaledObject.swift:24:46: 'characters' is deprecated: Please use String or Substring directly`, could you change that line to `let pathComponents = key.stringValue.split(separator: ".").map(String.init)`?

Reply:
I would branch off for 4 and 4.1 to prep for 5 changes just me and it seems to be how other libraries are doing this. Then the language version issues can be fixed and cocoapods/carthage make it easy to do deal with verson changes.